### ISSUE TITLE
Add JSON Schema

### ISSUE THREAD

**Issue by m-mohr** [stac-extensions/item-collection#1]

**Comment by matthewhanson** [comment]
@m-mohr I added an initial schema.json based on the original one from 0.9.0.

But appears the Node validator is already doing something with ItemCollections and says "not supported yet", guess it would need to be updated to handle the schema here.

```
- /home/matthew/work/stac/stac-extensions/item-collection/examples/itemcollection-sample-full.json
-- Lint: File is well-formed
-- The file is a /collections/:id/items endpoint. Validating all 1 items, but ignoring the other parts of the response.
-- CS3-20160503_132131_05: STAC Version: 1.0.0-rc.2
--- item: valid

- /home/matthew/work/stac/stac-extensions/item-collection/examples/itemcollection-sample-minimal.json
-- Lint: File is well-formed
-- STAC Version: 1.0.0-rc.2
-- Skipping; STAC ItemCollections not supported yet
```

**Comment by m-mohr** [comment]
Yes, indeed. Due to the missing JSON Schema, I had to short-circuit validation for Item Collections. It will require a new version to implement support for them.

**Comment by m-mohr** [comment]
The Schemas as is seems a bit strange though. The Item Collection could now implement e.g. STAC Version 0.9.0, but must contain STAC Items with version 1.0.0-rc.2. Also, to validate extensions for each item contained, we will still need to short circuit validation for item collections somehow, otherwise it will only be checked against the core item schema. So this is not trivial...

**Comment by matthewhanson** [comment]
Would it be easier if it were published as a core object?

**Comment by m-mohr** [comment]
No, we just don't have a structure yet, which consists of things that can have different stac_versions and extensions in an array as parts of the whole. That needs special handling in the validation tools. We can't solve that on the JSON Schema level.

**Comment by m-mohr** [comment]
It's the same for the CollectionCollection, which we should also write an extension for, I guess?!

**Comment by matthewhanson** [comment]
I don't think it's very high priority. The main, and currently only, use case is for the results of searches which have been constructed from, presumably validated, STAC metadata from an API.
